packages/neo-one-monitor/src/BrowserMetricsFactory.js

```diff
@@ -0,0 +1,149 @@
+/* @flow */
+import { utils } from '@neo-one/utils';
+
+import type {
+  MetricsFactory,
+  MetricLabels,
+  MetricConstruct,
+} from './MonitorBase';
+
+import type {
+  CounterMetric,
+  GaugeMetric,
+  HistogramMetric,
+  SummaryMetric,
+} from './types';
+
+let metrics = {
+  counters: {},
+  histograms: {},
+};
+
+export const resetMetricsForTesting = (): void => {
+  metrics = {
+    counters: {},
+    histograms: {},
+  };
+};
+
+export type MetricValue = {|
+  labels?: MetricLabels,
+  count?: number,
+|};
+
+export class CollectingMetric {
+  metric: MetricConstruct;
+  values: Array<MetricValue>;
+
+  constructor(metric: MetricConstruct) {
+    this.metric = metric;
+    this.values = [];
+  }
+
+  reset(): void {
+    this.values = [];
+  }
+}
+
+export type MetricCollection = {|
+  counters: { [name: string]: CollectingMetric },
+  histograms: { [name: string]: CollectingMetric },
+|};
+
+class BrowserCounter extends CollectingMetric implements CounterMetric {
+  inc(labels?: MetricLabels, count?: number): void {
+    this.values.push({ labels, count });
+  }
+}
+
+export class BrowserGauge extends CollectingMetric implements GaugeMetric {
+  // eslint-disable-next-line
+  inc(labels?: MetricLabels, count?: number): void {}
+
+  // eslint-disable-next-line
+  dec(labels?: MetricLabels, count?: number): void {}
+
+  // eslint-disable-next-line
+  set(labels?: MetricLabels, count?: number): void {}
+}
+
+class BrowserHistogram extends CollectingMetric implements HistogramMetric {
+  observe(labels?: MetricLabels, count?: number): void {
+    this.values.push({ labels, count });
+  }
+}
+
+export class BrowserSummary extends CollectingMetric implements SummaryMetric {
+  // eslint-disable-next-line
+  observe(labels?: MetricLabels, count?: number): void {}
+}
+
+export default class BrowserMetricsFactory implements MetricsFactory {
+  collect(): MetricCollection {
+    const currentMetrics = this._serializeJSON(metrics);
+    utils
+      .keys(metrics)
+      .map(metricType =>
```

Comment:
This would be an appropriate time to use `forEach` since you're doing a side effect.